packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart

```diff
@@ -168,26 +169,26 @@ class User extends UserInfo<auth_interop.UserJsImpl> {
   /// The user's phone number is in E.164 format (e.g. +16505550101).
   Future<ConfirmationResult> reauthenticateWithPhoneNumber(
           String phoneNumber, ApplicationVerifier applicationVerifier) =>
-      handleThenable(auth_interop.reauthenticateWithPhoneNumber(
-              jsObject, phoneNumber, applicationVerifier.jsObject))
-          .then(ConfirmationResult.fromJsObject);
+      auth_interop
+          .reauthenticateWithPhoneNumber(
+              jsObject, phoneNumber.toJS, applicationVerifier.jsObject)
+          .toDart as Future<ConfirmationResult>;
 
   /// Reauthenticates a user with the specified provider using
   /// a pop-up based OAuth flow.
   /// It returns the [UserCredential] information if reauthentication is successful.
   Future<UserCredential> reauthenticateWithPopup(AuthProvider provider) =>
-      handleThenable(
-              auth_interop.reauthenticateWithPopup(jsObject, provider.jsObject))
-          .then(UserCredential.fromJsObject);
+      auth_interop.reauthenticateWithPopup(jsObject, provider.jsObject).toDart
+          as Future<UserCredential>;
 
   /// Reauthenticates a user with the specified OAuth [provider] using
   /// a full-page redirect flow.
-  Future<void> reauthenticateWithRedirect(AuthProvider provider) =>
-      handleThenable(
-          auth_interop.reauthenticateWithRedirect(jsObject, provider.jsObject));
+  Future<void> reauthenticateWithRedirect(AuthProvider provider) => auth_interop
+      .reauthenticateWithRedirect(jsObject, provider.jsObject)
+      .toDart;
 
   /// If signed in, it refreshes the current user.
-  Future<void> reload() => handleThenable(jsObject.reload());
+  Future<void> reload() => jsObject.reload().toDart;
```

Comment:
I see `as Future<void>` isn't needed on other API with this type so perhaps it isn't needed like the others returning a type as opposed to void.